var/spack/repos/builtin/packages/py-msal-extensions/package.py

```diff
@@ -16,9 +16,13 @@ class PyMsalExtensions(PythonPackage):
     homepage = "https://github.com/AzureAD/microsoft-authentication-library-for-python"
     pypi = "msal-extensions/msal-extensions-0.2.2.tar.gz"
 
+    version("1.0.0", sha256="c676aba56b0cce3783de1b5c5ecfe828db998167875126ca4b47dc6436451354")
     version("0.2.2", sha256="31414753c484679bb3b6c6401623eb4c3ccab630af215f2f78c1d5c4f8e1d1a9")
     version("0.1.3", sha256="59e171a9a4baacdbf001c66915efeaef372fb424421f1a4397115a3ddd6205dc")
 
+    # https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/setup.py
     depends_on("py-setuptools", type="build")
     depends_on("py-msal@0.4.1:1", type=("build", "run"))
-    depends_on("py-portalocker@1.0:1", type=("build", "run"))
+    depends_on("py-portalocker@1.0:1", type=("build", "run"), when="@:0")
+    # This is the earliest version to work for Windows and Linux
+    depends_on("py-portalocker@1.6:1", type=("build", "run"), when="@1:")
```

Comment:
Can you move the `when` args before the `type`?  It aids readability IMO, e.g., "depends on py-portalocker at 1.6 through 1 when at (version) 1 on".

Reply:
I do this too, but only because the `when` arg comes before `type` in the function definition. Obviously these are all style things, anything technically works.